simulation/tests/test_agents.py

```diff
@@ -110,3 +113,42 @@ def test_household_is_renovating_state_updates(
         assert not household.is_renovating
         household.evaluate_renovation(model)
         assert household.is_renovating
+
+    def test_household_flat_without_roof_cannot_upgrade_roof_insulation(
+        self,
+    ) -> None:
+
+        household = household_factory(roof_energy_efficiency=np.NaN)
+        assert Element.ROOF not in household.get_upgradable_insulation_elements()
+
+    def test_household_elements_under_max_energy_efficiency_score_are_upgradable(
+        self,
+    ) -> None:
+
+        MAX_ENERGY_EFFICIENCY_SCORE = 5
```

Comment:
Found this a little confusing because you have `MAX_ENERGY_EFFICIENCY_SCORE` here but then there's `MAX_ENERGY_EFFICIENCY_SCORE` in the body of `get_upgradable_insulation_elements` too. What are you trying to get across?

Reply:
That 5 is the maximum score in the source dataset. I will update.